commands/charm/confirm/confirm.go

```diff
@@ -71,18 +88,31 @@ var promptTemplate = `
 `
 
 var resultTemplate = `
+{{- if RenderResult .FinalValue -}}
 {{- if.FinalValue -}}{{success promptPrefix}}{{else}}{{error promptPrefix}}{{end}}
 {{- print " " .Prompt " " -}}
-{{- if .FinalValue -}}
-	{{- success "yes" -}}
-{{- else -}}
-	{{- error "no" -}}
-{{- end }}
+	{{- if .FinalValue -}}
+		{{- success "yes" -}}
+	{{- else -}}
+		{{- error "no" -}}
+	{{- end }}
+{{- end -}}
 `
 
 func (p *Prompt) Prompt() (Choice, error) {
 	input := confirmation.New(p.text, fromChoice(p.choice))
-	input.ExtendedTemplateFuncs = charm.TemplateFuncs(charm.Colors)
+	tfs := template.Funcs(charm.Colors)
+	tfs["RenderResult"] = func(finalValue bool) bool {
+		switch p.displayResult {
+		case DisplayResultEphemeralNo:
+			return finalValue
+		case DisplayResultEphemeralYes:
+			return !finalValue
```

Comment:
I'm not quite sure if I followup this display result / finalValue interaction. Can you explain it (and maybe as a comment in the code)

Reply:
Yep I'll add some comments and see if I can come up with a more intuitive name. TL;DR DisplayResultEphemeralNo == ephemeral only if the choice is `no` i.e. the prompt disappears, otherwise it remains.

Reply:
92acda4cb6f1ab7b658305d9f6ffa4d04b029f95